lib/EventsGenerator.js

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2022 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+const utils = require('./utils')
+const eventsSdk = require('@adobe/aio-lib-events')
+const config = require('@adobe/aio-lib-core-config')
+const { getToken } = require('@adobe/aio-lib-ims')
+const { CLI } = require('@adobe/aio-lib-ims/src/context')
+
+const ActionGenerator = require('./ActionGenerator')
+const { promptForEventsOfInterest, getProviderMetadataToProvidersExistingMap } = require('./events/EventsOfInterestHelper')
+const { promptForRuntimeAction } = require('./events/RuntimeActionForEventsHelper')
+const eventsProvidermetadataToProviderMapping = 'AIO_events_providermetadata_to_provider_mapping'
+
+class EventsGenerator extends ActionGenerator {
+  constructor (args, opts) {
+    super(args, opts)
+    this.option('skip-prompt', { default: false }) // prompt to ask action name
+    // required
+    this.option('config-path',
+      { type: String, description: 'relative path to the config yaml file' })
+    this.option('full-key-to-events-manifest', {
+      type: String,
+      description: 'key in config path that resolves to events manifest e.g. "application.events"',
+      default: ''
+    })
+
+    this.configPath = this.destinationPath(this.options['config-path'])
+    this.fullKeyToRuntimeManifest = this.options['full-key-to-manifest']
+    this.actionFolder = this.options['action-folder'] // todo ensure this is relative to root
```

Comment:
are we leaving todos in here, was this overlooked? if not file an issue to take care of it later

Reply:
this is a TODO in action generator as well. so if it needs to be taken care of there, added a comment here so that it is not missed here